xray/utils/downloadindexer.go

```diff
@@ -21,6 +25,8 @@ import (
 const (
 	indexerFileName     = "indexer-app"
 	graphScanMinVersion = "3.29.0"
+	indexerDirName      = "xray-indexer"
+	tempIndexerDirName  = "temp"
 )
 
 func DownloadIndexerIfNeeded(xrayManager *xray.XrayServicesManager) (string, error) {
```

Comment:
If multiple jfrog-cli processes are running on the same machine, some of them may attempt to download the indexer at rthe same time, which can lead to issues.
I therefore suggest putting in place a cross-process lock on this function. We've already implemented such a locking mechanism on the config, which has been serving us well for a long time , so you can reuse this locking mechism.